nemo/collections/tts/torch/data_ssl_vocoder.py

```diff
@@ -0,0 +1,624 @@
+import json
+import os
+import pickle
+import shutil
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import hydra.utils
+import librosa
+import numpy as np
+import torch
+from omegaconf import open_dict
+from tqdm import tqdm
+
+from nemo.collections.asr.parts.preprocessing.features import WaveformFeaturizer
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.tts.models import ssl_tts
+from nemo.collections.tts.torch.helpers import get_base_dir
+from nemo.collections.tts.torch.tts_tokenizers import EnglishCharsTokenizer
+from nemo.core.classes import Dataset
+from nemo.utils import logging
+
+
+class SSLVocoderDataset(Dataset):
+    def __init__(
+        self,
+        manifest_filepath: Union[str, Path, List[str], List[Path]],
+        sample_rate: int,
+        ssl_model_ckpt_path: Union[str, Path],
+        ssl_content_emb_type: str,
+        max_duration: Optional[float] = None,
+        min_duration: Optional[float] = None,
+        ssl_downsampling_factor: Optional[int] = 4,
+        ignore_file: Optional[Union[str, Path]] = None,
+        trim: Optional[bool] = False,
+        pitch_conditioning: Optional[bool] = False,
+        pitch_mean: Optional[float] = None,
+        pitch_std: Optional[float] = None,
+        pitch_normalization: Optional[str] = None,
+        sup_data_dir: Optional[Union[str, Path]] = None,
+        recache_data: Optional[bool] = False,
+        normalize_content: Optional[bool] = True,
+        speaker_stats_pitch_fp: Optional[Union[str, Path]] = None,
+        use_unique_tokens: Optional[bool] = False,
+        speaker_conditioning_type: Optional[str] = "per_sample",  # per_sample, mean, interpolate,
+        segment_speaker_embedding: Optional[bool] = False,
+    ):
+        """Dataset used for training FastPitchModel_SSL model.
+        Args:
+            manifest_filepath (Union[str, Path, List[str], List[Path]]): Path(s) to the .json manifests containing
+            information on the dataset. Each line in the .json file should be valid json. Note: the .json file itself
+            is not valid json. Each line should contain the following:
+                "audio_filepath": <PATH_TO_WAV>,
+                "speaker" : <SPEAKER NUM>
+                "duration": <Duration of audio clip in seconds> (Optional)
```

Comment:
Just to confirm--the manifest has the same fields as usual?

Reply:
yes, I am pretty sure.